### DISCUSSION TITLE
Delay in event capturing by Quorum chaincode event listener

### DISCUSSION THREAD

**Discussion by Purbaja** [ConsenSys/quorum#1603] [Tooling]
We are observing delay between event emission and event being captured by the chaincode event listener.

Transaction (which emits event) was posted from Application at - 11:40:44 GMT
Block mining timestamp for this txn was: 1670931644873193200  i.e. 11:40:44 GMT
Event captured by the chaincode event listener at – 11:40:57 GMT

So, what could be the reason for the listener to take 13sec to read the event?

Quorum network overview:
No of nodes in the network : 20
Block height : 46016536

**Comment by baptiste-b-pegasys** [comment]
Are you using `LogFilterer`?

Events depends on the node you register the subscription. Is this node was syncing at that moment?

It also depends also on the number of events you are processing, the client buffers up to 20k events. It could be on the queue until it's been processed.

Can you provide more details how you achieved this or a way to reproduce this?

**Comment by sabiriqbal** [comment]
1. We are using EthFilter not LogFilterer. and passing that filter to web3j.ethLogFlowable(filter).
2. Yes node is syncing continuously.
